metadata/metadata.go

```diff
@@ -886,6 +1056,46 @@ func (resource *trainingSetVariantResource) Update(lookup ResourceLookup, update
 	return nil
 }
 
+func (resource *trainingSetVariantResource) IsEquivalent(other ResourceVariant) bool {
+	/**
+	Key Fields for a TrainingSetVariant are
+	- Name
+	- Labels
+	- Features
+	- Lag Features
+	- Owner
+	*/
+	otherVariant, ok := other.(*trainingSetVariantResource)
+	if !ok {
+		return false
+	}
+
+	thisProto := resource.serialized
+	otherProto := otherVariant.serialized
+
+	if otherProto.Status.Status == pb.ResourceStatus_FAILED {
```

Comment:
what should happen if a resource status is PENDING?

Reply:
we just create another variant right now